contracts/cryptoblades.sol

```diff
@@ -711,24 +711,24 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
         }
     }*/
 
-    function mintWeaponN(uint32 num)
+    function mintWeaponN(uint32 num, uint16 chosenElement, uint16 chosenElementFee)
```

Comment:
Needs check on chosenElementFee value to prevent passing 0 as chosenElementFee there and minting weapons for free with `line 720:  _payContract(msg.sender, mintWeaponFee * num * 0);`

Use e.g. `require(chosenElementFee > <whatever the required fee is>)`